lib/Parser/Parse.cpp

```diff
@@ -5040,6 +5040,17 @@ ParseNode * Parser::ParseFncDecl(ushort flags, LPCOLESTR pNameHint, const bool n
         }
     }
 
+    if (fDeclaration)
+    {
+        Symbol* funcSym = pnodeFnc->GetFuncSymbol();
+        if (funcSym != nullptr &&
+            funcSym->GetIsFormal() &&
+            GetCurrentFunctionNode() != nullptr)
+        {
```

Comment:
If the sym is a formal, then the current function node must be non-null, since the global code can't have formals.